### PR TITLE
feat(NoDebtCertificate): New no debt certificate application

### PR DESCRIPTION
**Pull request by jonbjarnio**
# ...

## What

Creating new no debt certificate application

## Why

So people can get their no debt certificates

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/application/templates/no-debt-certificate/src/fields/ConfirmationPendingField/index.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+export const ConfirmationPendingField: FC<FieldBaseProps> = ({
+  application,
+  refetch,
+}) => {
+  const applicationId = application.id
+
+  const [finishSubmit, setFinishSubmit] = useState<boolean>(false)
+
+  const [submitApplication] = useMutation(SUBMIT_APPLICATION, {
+    onError: (e) => console.error(e.message),
```
Comment (saevarma) [inline libs/application/templates/no-debt-certificate/src/fields/ConfirmationPendingField/index.tsx:16]: Shouldn't this use the `handleServerError` utility in `application/ui-components`?

File: libs/application/templates/no-debt-certificate/src/lib/noDebtCertificateTemplate.ts [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+          },
+          actionCard: {
+            tag: {
+              label: m.actionCardDone,
+              variant: 'blueberry',
+            },
+          },
+          // onEntry: {
+          //   apiModuleAction: ApiActions.getNoDebtCertificate,
+          // },
```
Comment (saevarma) [inline libs/application/templates/no-debt-certificate/src/lib/noDebtCertificateTemplate.ts:91]: Remove redundant comments

File: libs/application/templates/no-debt-certificate/src/shared/constants.ts [hunk 3]
```diff
@@ -0,0 +1,2 @@
+export enum ApiActions {}
+//getNoDebtCertificate = 'getNoDebtCertificate',
```
Comment (saevarma) [inline libs/application/templates/no-debt-certificate/src/shared/constants.ts:2]: Remove redundant comment

File: libs/application/templates/no-debt-certificate/src/dataProviders/NoDebtCertificateProvider.ts [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  SuccessfulDataProviderResult,
+  FailedDataProviderResult,
+} from '@island.is/application/core'
+import { m } from '../lib/messages'
+import { MessageDescriptor } from '@formatjs/intl'
+
+export class NoDebtCertificateProvider extends BasicDataProvider {
+  type = 'NoDebtCertificateProvider'
+
+  async provide(): Promise<unknown> {
```
Comment (saevarma) [inline libs/application/templates/no-debt-certificate/src/dataProviders/NoDebtCertificateProvider.ts:13]: The return type should preferably by typed

**Review by saevarma — CHANGES_REQUESTED** [review]
Few comments and shouldn't the Driving School related things be separate?

File: libs/application/templates/driving-school-confirmation/src/fields/NationalIdWithName/index.tsx [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+const IdentityQuery = gql`
+  query IdentityQuery($input: IdentityInput!) {
+    identity(input: $input) {
+      name
+      nationalId
+    }
+  }
+`
+
+const NationalIdWithName: FC<FieldBaseProps> = ({ field, application }) => {
```
Comment (albinagu) [inline libs/application/templates/driving-school-confirmation/src/fields/NationalIdWithName/index.tsx:22]: im wondering why these changes are in here? this is something thats been recently merged to main in another PR
